mbed_greentea/mbed_test_api.py

```diff
@@ -174,12 +174,28 @@ def run(self):
 
         def stop(self):
             self.active = False
+
+            # Try stopping mbed-host-test
             try:
-                self.proc.terminate()
-            except Exception as e:
-                print "ProcessObserver.stop(): %s" % str(e)
+                self.proc.stdin.close()
+            finally:
                 pass
 
+            # Give 5 sec for mbedhtrun to exit
+            for i in range(5):
+                ret_code = self.proc.poll()
+                if ret_code is not None:
```

Comment:
Please add comment here:
```python
#  A None value indicates that the process hasn’t terminated yet.
```
so it would be easier to understand this code (why you just didn't write
```python
if ret_code:
   ...
```